docs/adr/006_cups.md

```diff
@@ -0,0 +1,285 @@
+
+### Facade Azure Function related changes
+
+The Azure Function should implement a new endpoint 'FetchConcentratorCredentials' which executes the following flow:
+
+```mermaid
+sequenceDiagram
+LoRaWanNetworkSrvModule->>Function: POST /fetchConcentratorCredentials
+Note right of LoRaWanNetworkSrvModule: Request includes 'stationEui' and 'credentialsType' (cups/tc/both)
+Function->>IoT Hub: Retrieve Concentrator device Twin
+Function->>Function: Parse *CredentialUrl string to understand credentials location
+alt *CredentialUrl is a KeyVault Secret
+    Function->>Azure KeyVault: Get KeyVault secret using Azure Function Managed Identity
+else *CredentialUrl is from Azure Blob Storage and no SAS is in Url
+    Function->>Azure Blob Storage: Get Blob using MSI
+else *CredentialUrl is plain Url
+    Function->>HTTPS Endpoint: Get Blob with HTTP GET Request
+end
+Function->>LoRaWanNetworkSrvModule: Return credential blob
+Note right of LoRaWanNetworkSrvModule: Response should be a json including base64 encoded credential
+
+```
+
+In addition to this, the Azure Function must be able to properly authenticate to target sink via Managed Identity (when using Azure Blob Storage or KeyVault).
+
+Considering that the default sink for the starter kit should be an Azure Storage Account, the template for the starter kit should be changed in such a way that a new Blob Container is created for uploading credential blob files.
+
+If using KeyVault secrets instead of Blob Storage, instructions on how to create a role assignment should be provided.  
+
+### LoRaWan Network Server changes
+
+- LoRaDeviceAPIServiceBase and its implementation (LoRaDeviceAPIService) should include a method for calling the 'FetchConcentratorCredentials' endpoint
+- IBasicsStationConfigurationService and its implementation (BasicsStationConfigurationService) should include a method for retrieving and parsing "cups" desired property in concentrator device twin
+- BasicsStationNetworkServerStartup should include a new endpoint for handling CUPS '/update-info' POST requests
+- BasicsStationNetworkServer should implement a basic client certificate validation when configuring kestrel
+- A new CupsProtocolMessageProcessor class is implemented for achieving purposes of above drawn sequence diagram.
+
+### LoRa Device Provisioning CLI changes
+
+The CLI should be changed in order to:
+
+- Specify if the device being created is making use of CUPS or not
+- In case a device being created is making use of CUPS, the tool should:
+  - accept as input a cupsCredentials file (being the concatenation of `cups.{trust,cert,key}`)
+  - accept as input a tcCredentials file (being the concatenation of `tc.{trust,cert,key}`)
+  - accept as input a clientThumbprint string (for client certificate validation)
+  - compute the CRC32 of above mentioned files
+  - create a device for the concentrator
```

Comment:
Maybe it would be worth mentioning certs rotation

Reply:
Added a section already. Would you mind having a look now? @bastbu

Reply:
I saw that I had the comment twice. Sorry!